libs/core/src/main/java/org/elasticsearch/core/internal/io/Streams.java

```diff
@@ -56,4 +56,21 @@ public static long copy(final InputStream in, final OutputStream out) throws IOE
         }
     }
 
+    /**
+     * Returns a temporary thread local buffer.
+     */
+    public static byte[] getTemporaryBuffer() {
```

Comment:
Can we do without this method? I think exposing a buffer from this class could be error prone